### PR TITLE
Adding jenkins file for backing up indexes to S3

### PR DESCRIPTION
**Pull request by shgriffi**
This is for https://projects.engineering.redhat.com/browse/DATAHUB-499.  It was tested in minishift but not Jenkins in upshift yet.

The accompanying commits to the elasticsearch repo that installs the s3 plugin to elasticsearch are as follows:

https://github.com/shgriffi/origin-aggregated-logging/commit/f2fbf15f06fb04667d84ecfc33323d9202c1a911

https://github.com/shgriffi/origin-aggregated-logging/commit/64e905434a8f1f7a8f89cc563cd5bd2b52baa4b7

### CODE REVIEW

**Comment by t0ffel** [comment]
There is no need to have this in openshift-ansible repo as this doesn't use any of the roles. It is better to create a new small repo with the backup stuff, but origin-aggregated-logging stuff needs to be merged.

File: ci/stage/elasticsearch_s3backup.Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      ],
+      submoduleCfg: [],
+      userRemoteConfigs: [[url: 'https://gitlab.cee.redhat.com/data-hub/dh-ci-util.git']]
+    ]
+}
+def start_backup() {
+  try {
+    // Check if repository exists
+    checkRepositoryResponse = sh (
+      script: 'curl --insecure --cert $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.crt --key $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.key -i -k ${ELASTICSEARCH_ENDPOINT}/_snapshot/${BACKUP_NAME}',
```
Comment (t0ffel) [inline ci/stage/elasticsearch_s3backup.Jenkinsfile:38]: where are the variables `ELASTICSEARCH_ENDPOINT`, `BACKUP_NAME` coming from?

File: ci/stage/elasticsearch_s3backup.Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      returnStdout: true
+    ).trim()
+    
+    repoDoesNotExist = checkRepositoryResponse.toLowerCase().contains('404 not found')
+    
+    // Create the repository if one does not exist
+    if (repoDoesNotExist)
+    {
+        createRepositoryResponse = sh (
+            script: 'curl --insecure --cert $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.crt --key $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.key -X PUT -H "Content-Type: application/json" -d ' + /'{"type": "s3","settings": {"bucket": "${S3_BUCKET}","base_path": "${S3_PATH}","endpoint": "${S3_ENDPOINT}","access_key":"${S3_ACCESS_KEY}","secret_key":"${S3_SECRET_KEY}"}}'/ + ' -i ${ELASTICSEARCH_ENDPOINT}/_snapshot/${BACKUP_NAME}',
```
Comment (t0ffel) [inline ci/stage/elasticsearch_s3backup.Jenkinsfile:48]: also where are all the env vars coming from?

File: ci/stage/elasticsearch_s3backup.Jenkinsfile (hunk 1 above)
Comment (shgriffi) [reply ci/stage/elasticsearch_s3backup.Jenkinsfile:38]: These will be passed in as parameters in Jenkins.  We can create a story for the JJB part of it.

File: ci/stage/elasticsearch_s3backup.Jenkinsfile (hunk 2 above)
Comment (shgriffi) [reply ci/stage/elasticsearch_s3backup.Jenkinsfile:48]: Which environment variables? The S3 ones are going to be passed in via Jenkins.  There will be default values for them.

**Comment by shgriffi** [comment]
Do we need to move this to a new repo now?  Can that be done at a later date?

**Pull request closed without merging**